orchagent/high_frequency_telemetry/hftelorch.cpp

```diff
@@ -0,0 +1,729 @@
+        return task_process_status::task_success;
+    }
+
+    profile->tryCommitConfig(type);
+
+    m_type_profile_mapping[type].insert(profile);
+
+    SWSS_LOG_NOTICE("The high frequency telemetry group %s with profile %s is set (object_names: %s, object_counters: %s)",
+                    group_name.c_str(),
+                    profile_name.c_str(),
+                    arg_object_names ? arg_object_names->c_str() : "",
+                    arg_object_counters ? arg_object_counters->c_str() : "");
+
+    return task_process_status::task_success;
+}
+
+task_process_status HFTelOrch::groupTableDel(const std::string &profile_name, const std::string &group_name)
+{
+    SWSS_LOG_ENTER();
+
+    auto profile = tryGetProfile(profile_name);
+
+    if (!profile)
+    {
+        SWSS_LOG_WARN("The high frequency telemetry profile %s is not found", profile_name.c_str());
+        return task_process_status::task_success;
+    }
+
+    auto type = HFTelUtils::group_name_to_sai_type(group_name);
+
+    if (!profile->canBeUpdated(type))
+    {
+        return task_process_status::task_need_retry;
+    }
+
+    profile->clearGroup(group_name);
+    m_type_profile_mapping[type].erase(profile);
+    m_state_telemetry_session.del(profile_name + "|" + group_name);
+
+    SWSS_LOG_NOTICE("The high frequency telemetry group %s with profile %s is deleted", group_name.c_str(), profile_name.c_str());
+
+    return task_process_status::task_success;
+}
+
+shared_ptr<HFTelProfile> HFTelOrch::getProfile(const string &profile_name)
+{
+    SWSS_LOG_ENTER();
+
+    if (m_name_profile_mapping.find(profile_name) == m_name_profile_mapping.end())
+    {
+        m_name_profile_mapping.emplace(
+            profile_name,
+            make_shared<HFTelProfile>(
+                profile_name,
+                m_sai_tam_obj,
+                m_sai_tam_collector_obj,
+                m_counter_name_cache));
+    }
+
+    return m_name_profile_mapping.at(profile_name);
+}
+
+std::shared_ptr<HFTelProfile> HFTelOrch::tryGetProfile(const std::string &profile_name)
+{
+    SWSS_LOG_ENTER();
+
+    auto itr = m_name_profile_mapping.find(profile_name);
+    if (itr != m_name_profile_mapping.end())
+    {
+        return itr->second;
+    }
+
+    return std::shared_ptr<HFTelProfile>();
+}
+
+void HFTelOrch::doTask(swss::NotificationConsumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    std::string op;
+    std::string data;
+    std::vector<swss::FieldValueTuple> values;
+
+    if (&consumer != m_asic_notification_consumer.get())
+    {
+        SWSS_LOG_DEBUG("Is not TAM notification");
+        return;
+    }
+
+    consumer.pop(op, data, values);
+
+    if (&consumer != m_asic_notification_consumer.get())
+    {
+        SWSS_LOG_ERROR("Unknown consumer");
+        return;
+    }
+
+    if (op != SAI_SWITCH_NOTIFICATION_NAME_TAM_TEL_TYPE_CONFIG_CHANGE)
+    {
+        SWSS_LOG_ERROR("Unknown operation type %s", op.c_str());
+        return;
+    }
```

Comment:
I don't think this should be check on NotificationConsumer doTask, I see that it prints error when port is shutdown:
`ERR swss#orchagent: :- doTask: Unknown operation type port_state_change`